src/thorlabs_cube/driver/tcube/tsc.py

```diff
@@ -0,0 +1,407 @@
+        """
+
+        # Send the message with the command ID (0x0465), and put channel ID in param1 and stop mode in param2
+        await self.send_request(
+            "<BB",
+            MGMSG.MOT_MOVE_STOP,
+            [MGMSG.MOT_MOVE_STOPPED],
+            param1=_CHANNEL,
+            param2=stop_mode,
+        )
+
+    async def set_av_modes(self, mode_bits):
+        """Set the LED indicator modes based on the provided mode_bits.
+
+        :param mode_bits: A bitmask indicating which modes to enable:
+                        - 1 (LEDMODE_IDENT): LED flashes when 'Ident' is sent.
+                        - 2 (LEDMODE_LIMITSWITCH): LED flashes when motor reaches limit switch.
+                        - 8 (LEDMODE_MOVING): LED is lit when the motor is moving.
+        """
+
+        # Pack the Channel ID (2 bytes) and ModeBits (4 bytes)
+        payload = st.pack("<Hl", 1, mode_bits)
+
+        # Send the message with the packed payload for MOT_SET_AVMODES (0x04B3)
+        await self.send(Message(MGMSG.MOT_SET_AVMODES, data=payload))
+
+    async def get_av_modes(self):
+        """Get the current LED indicator mode bits.
+
+        :return: The LED mode bits.
+        """
+
+        # Send the request to get the current AV modes for the specified channel
+        get_msg = await self.send_request(
+            MGMSG.MOT_REQ_AVMODES, [MGMSG.MOT_GET_AVMODES], 1
+        )
+
+        mode_bits = get_msg.data[2:]
+        return mode_bits
+
+    async def set_button_parameters(
+        self, mode, position1, position2, timeout1, timeout2
+    ):
+        """Set button parameters for the front panel buttons.
+
+        :param mode: Mode for the buttons (1 for jog, 2 for position mode).
+        :param position1: Position in encoder counts for the top button.
+        :param position2: Position in encoder counts for the bottom button.
+        :param timeout1: Timeout in ms for position1.
+        :param timeout2: Timeout in ms for position2.
+        """
+
+        # Pack the channel ID, mode, position1, position2, timeout1, and timeout2
+        payload = st.pack("<HHllHH", 1, mode, position1, position2, timeout1, timeout2)
+
+        # Send the message with the packed payload for MOT_SET_BUTTONPARAMS (0x04B6)
+        await self.send(Message(MGMSG.MOT_SET_BUTTONPARAMS, data=payload))
+
+    async def get_button_parameters(self):
+        """Get the current button parameters.
+
+        :return: A tuple containing (mode, position1, position2, timeout1, timeout2)
+        """
+
+        # Send the request for button parameters (MOT_REQ_BUTTONPARAMS 0x04B7)
+        get_msg = await self.send_request(
+            MGMSG.MOT_REQ_BUTTONPARAMS, [MGMSG.MOT_GET_BUTTONPARAMS], 1
+        )
+
+        mode, position1, position2, timeout1, timeout2 = st.unpack(
+            "<HllHH", get_msg.data[2:]
+        )
+
+        return mode, position1, position2, timeout1, timeout2
```

Comment:
```suggestion
        st.unpack("<HllHH", get_msg.data[2:])
```
No need to unpack the tuple yourself here.